solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,530 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "@bancor/token-governance/contracts/ITokenGovernance.sol";
+
+import "../converter/interfaces/IConverterRegistry.sol";
+import "../converter/interfaces/IConverter.sol";
+
+import "../utility/ContractRegistryClient.sol";
+import "../utility/Owned.sol";
+import "../utility/Utils.sol";
+import "../utility/TokenHolder.sol";
+import "../utility/ReentrancyGuard.sol";
+
+import "../INetworkSettings.sol";
+import "../BancorNetwork.sol";
+
+/**
+ * @dev This contract provides any user to perform a vortex
+ */
+contract VortexBurner is Owned, Utils, ReentrancyGuard, ContractRegistryClient {
+    using SafeMath for uint256;
+    using Math for uint256;
+    using SafeERC20 for IERC20;
+
+    struct Strategy {
+        address[][] paths;
+        uint256[] amounts;
+        address[] govPath;
+    }
+
+    struct NetNetworkTokenConversionAmounts {
+        uint256 amount;
+        uint256 incentiveFeeAmount;
+    }
+
+    // the vortex is only designed to work with 50/50 standard pool converters
+    uint32 private constant STANDARD_POOL_RESERVE_WEIGHT = PPM_RESOLUTION / 2;
+
+    // the type of the standard pool converter
+    uint16 private constant STANDARD_POOL_CONVERTER_TYPE = 3;
+
+    // the address of the network token
+    IERC20 private immutable _networkToken;
+
+    // the address of the governance token
+    IERC20 private immutable _govToken;
+
+    // the address of the governance token security module
+    ITokenGovernance public immutable _govTokenGovernance;
+
+    // the percentage of the converted network tokens to be sent to the caller of the vortex (in units of PPM)
```

Comment:
generally speaking, `vortex` is the name of the mechanism, `burner` is the name of this contract, `burning event` is the actual event/action triggered by the user (might be a better name for this one). it makes the comments/terminology in a bunch of places in the contract a bit confusing ("completed vortex")

Reply:
I'll rename

Reply:
I think the `vortex` function also needs a different name. maybe `burn`?

Reply:
👍